hazelcast/src/main/java/com/hazelcast/util/HashUtil.java

```diff
@@ -334,12 +334,10 @@ public static int hashToIndex(int hash, int length) {
         checkPositive(length, "length must be larger than 0");
 
         if (hash == Integer.MIN_VALUE) {
-            hash = 0;
-        } else {
-            hash = abs(hash);
+            return 0;
         }
 
-        return hash % length;
+        return abs(hash) % length;
```

Comment:
According to [Math#abs javaDoc](https://docs.oracle.com/javase/6/docs/api/java/lang/Math.html#abs(int))  `Integer.MIN_VALUE` is a special case and returns itself. Doesn't this change break the behaviour of the method?

Reply:
Oops I see now, you are right.